components/TipButton.tsx

```diff
@@ -0,0 +1,125 @@
+import { useState, useEffect } from 'react';
+import Image from 'next/image';
+import { Button } from '@/components/ui/button';
+import { useTipProvider } from '@/providers/TipProvider';
+import { SupabasePost } from '@/types/SupabasePost';
+import { useNeynarProvider } from '@/providers/NeynarProvider';
+import { Popover, PopoverContent } from '@/components/ui/popover';
+import { cn, isValidNumber } from '@/lib/utils';
+import { PopoverTrigger } from '@radix-ui/react-popover';
+import { useUi } from '@/providers/UiProvider';
+import { Badge } from './ui/badge';
+import { Input } from './ui/input';
+
+const defualtTips = {
+  DEGEN: [10, 50, 100],
+  NOTES: [100, 1000, 10000],
+};
+
+const logos = {
+  DEGEN: '/images/degenchain.png',
+  NOTES: '/images/notes.png',
+};
+
+export default function TipButton({
+  verifications,
+  cast = {} as SupabasePost,
+  currency,
+  className,
+}: {
+  verifications: string[];
+  cast: SupabasePost;
+  currency: 'DEGEN' | 'NOTES';
+  className?: string;
+}) {
+  const { user } = useNeynarProvider();
+  const userFid = user?.fid;
+  const castAuthorFid = cast.author?.fid;
+  const isSelfPost = userFid === castAuthorFid;
+  const { tip, tipDegen } = useTipProvider();
+  const [showDropdown, setShowDropdown] = useState(false);
+  const [customTip, setCustomTip] = useState('');
+  const [total, setTotal] = useState(0);
+  const { checkLoggedIn } = useUi();
+
+  useEffect(() => {
+    if (currency === 'NOTES') setTotal(cast.points ?? 0);
+    else if (currency === 'DEGEN') setTotal(cast.degen ?? 0);
+  }, [cast, currency]);
+
+  const handleTip = async (amount: number) => {
+    setShowDropdown(false);
+    if (amount === 0) return;
+    let response;
+    if (currency === 'NOTES') {
+      response = await tip(amount, cast.post_hash, cast.author.fid);
```

Comment:
please double check you are not modifying existing `tip` logic for either
- notes
- degen